BDArmory/Control/MissileFire.cs

```diff
@@ -2960,12 +2984,22 @@ public void UpdateList()
                     {
                         continue;
                     }
+                    if (weapon.Current.GetWeaponClass() == WeaponClasses.Missile || weapon.Current.GetWeaponClass() == WeaponClasses.Bomb || weapon.Current.GetWeaponClass() == WeaponClasses.SLW)
```

Comment:
Shouldn't this be inside the `if (!alreadyAdded)` block below, otherwise the list of ranges for a `weaponName` will have multiple entries of the same range?

Reply:
Technically, you could use `Dictionary<string, HashSet<float>>` for `rangeList` to avoid duplicates, but with the number of ranges per weapon type I think a list is more efficient due to the overhead of hashsets.
Also, a minor niggle, can we rename `rangeList` to `weaponRanges` to be consistent with `weaponTypes` and `weaponArray`?

Reply:
yes, it should be. Dictionary has been renamed for homogeneity with other UpdateList lists